backends/s3/credentials_test.go

```diff
@@ -0,0 +1,103 @@
+package s3_test
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/PowerDNS/simpleblob/backends/s3"
+	"github.com/PowerDNS/simpleblob/backends/s3/s3testing"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+func TestFileSecretsCredentials(t *testing.T) {
+	tempDir := t.TempDir()
+
+	// Instanciate provider (what we're testing).
+	provider := &s3.FileSecretsCredentials{
+		AccessKeyFilename: filepath.Join(tempDir, "access-key"),
+		SecretKeyFilename: filepath.Join(tempDir, "secret-key"),
+	}
+
+	// writeFiles creates or overwrites provider files
+	// with the same content.
+	writeFiles := func(content string) {
+		writeContent := func(filename string) {
+			f, err := os.Create(filename)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer f.Close()
+			if content == "" {
+				return
+			}
+			_, err = io.WriteString(f, content)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+		writeContent(provider.AccessKeyFilename)
+		writeContent(provider.SecretKeyFilename)
+	}
+
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// Create server
+	addr, stop, err := s3testing.ServeMinio(ctx, tempDir)
+	if errors.Is(err, s3testing.ErrMinioNotFound) {
+		t.Skip("minio binary not found locally, make sure it is in PATH")
+	}
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() { _ = stop() }()
+
+	// First credential files creation.
+	// Keep them empty for now,
+	// so that calls to the server will fail.
+	writeFiles("")
+
+	// Create minio client, using our provider.
+	creds := credentials.New(provider)
+	clt, err := minio.New(addr, &minio.Options{
```

Comment:
Instead of using a minio client for testing, it would be more useful to use the S3 simpleblob backend instead, since it would test the entire integration. This could also catch mistakes like using 'json' instead of 'yaml' in the struct tags.

Reply:
I wanted to implement the test at a lower level, to test the `minio.Provider` itself and not the backend, but indeed another test makes sense